test/unit/spec/common/jobCommChannel-spec.js

```diff
@@ -86,887 +86,962 @@ define([
+                        const comm = new JobCommChannel(),
+                            msgData = { [TEST_JOB_ID]: value },
+                            msg = makeCommMsg(msgType, msgData);
 
+                        return comm.initCommChannel().then(() => {
+                            spyOn(comm, 'reportCommMessageError');
+                            spyOn(console, 'error');
+                            comm.handleCommMessages(msg);
+                            expect(comm.reportCommMessageError.calls.allArgs()).toEqual([
+                                [{ msgType, msgData }],
+                            ]);
+                        });
+                    });
+                });
+
+                // message where all data is invalid
+                it(`should not send a ${msgType} message, many jobs, invalid`, () => {
+                    const msg = makeCommMsg(msgType, expected.invalid),
+                        comm = new JobCommChannel();
                     return comm.initCommChannel().then(() => {
                         spyOn(comm, 'reportCommMessageError');
                         spyOn(console, 'error');
                         comm.handleCommMessages(msg);
                         expect(comm.reportCommMessageError.calls.allArgs()).toEqual([
-                            [{ msgType, msgData }],
+                            [{ msgType, msgData: expected.invalid }],
                         ]);
                     });
                 });
+
+                // message with a mix of valid and invalid data
+                it(`should separate out valid and invalid ${type} data`, () => {
+                    const msg = makeCommMsg(msgType, validAndInvalidData),
+                        comm = new JobCommChannel();
+                    return comm.initCommChannel().then(() => {
+                        spyOn(comm, 'reportCommMessageError');
+                        spyOn(console, 'error');
+                        spyOn(testBus, 'send');
+                        comm.handleCommMessages(msg);
+                        expect(comm.reportCommMessageError.calls.allArgs()).toEqual([
+                            [{ msgType, msgData: expected.invalid }],
+                        ]);
+
+                        const expectedMessages = Object.keys(expected.valid).map((key) => {
+                            return [
+                                expected.valid[key],
+                                {
+                                    channel: { [JOB_CHANNEL]: key },
+                                    key: { type: msgType },
+                                },
+                            ];
+                        });
+
+                        expect(testBus.send.calls.allArgs()).toEqual(
+                            jasmine.arrayWithExactContents(expectedMessages)
+                        );
+                    });
+                });
+            });
+        });
+
+        describe('job mapping', () => {
```

Comment:
Content changes start here! Added some tests for the new mapping making functionality.